scripts/pre_commit_linter.py

```diff
@@ -1748,9 +1796,10 @@ def main():
     # debug mode which when enabled prints the tag_stack for each file.
     html_tag_and_attribute_messages = _check_html_tags_and_attributes(all_files)
     html_linter_messages = _lint_html_files(all_files)
-    linter_messages = _pre_commit_linter(all_files)
     pattern_messages = _check_bad_patterns(all_files)
     copyright_notice_messages = _check_for_copyright_notice(all_files)
+    linter_messages = _pre_commit_linter(all_files)
```

Comment:
Instead of changing this order, I recommend we move the "Summary of errors" section from _pre_commit_linter function to _print_summary_messages function.

Reply:
Or better so, move the _pre_commit_linter function to the top, call it before all other functions here -- this might be most efficient since it runs in parallel threads.

Reply:
done